### PR TITLE
feat(26217): Add missing member fix should work for missing function

### LINKED ISSUE #26217: Add missing member fix should work for missing function
**TypeScript Version:**  3.1.0-dev.20180804

**Code**

```ts
class C {}
new C().m();

f();

namespace N { export const x = 0; }
N.f();

import * as m from "./mod";
m.f();
```

**Expected behavior:**

Codefix for both `m()` and for every `f()`.

**Actual behavior:**

Codefix for only `m()`.

### PR DESCRIPTION
**Pull request by a-tarasyuk**
Fixes #26217

### CODE REVIEW

**Comment by DanielRosenwasser** [comment]
@typescript-bot pack this

File: src/services/codefixes/helpers.ts [hunk 1]
```diff
@@ -381,14 +391,18 @@ namespace ts.codefix {
             createStubbedMethodBody(quotePreference));
     }
 
-    function createStubbedMethodBody(quotePreference: QuotePreference): Block {
+    function createStubbedMethodBody(quotePreference: QuotePreference) {
+        return createStubbedBody("Method not implemented.", quotePreference);
```
Comment (sandersn) [inline src/services/codefixes/helpers.ts:395]: shouldn't this be a message in diagnosticMessages? I feel like people will want localised versions of it.

**Review by sandersn — APPROVED** [review]
Just a couple of early comments -- I've looked at the tests, and they look good, but haven't looked at the code.

File: src/services/codefixes/helpers.ts (hunk 1 above)
Comment (a-tarasyuk) [reply src/services/codefixes/helpers.ts:395]: Good question. I think you are right, I don't know why this message wasn't added to the Diagnostics before. Anyway I'll add the following messages to Diagnostics list

1. `"Method not implemented."`
2. `"Function not implemented."`

File: tests/cases/fourslash/importNameCodeFixNewImportNodeModules5.ts [hunk 2]
```diff
@@ -18,5 +18,14 @@
 ////   "types": "bin/lib/libfile.d.ts"
 //// }
 
-// No fix because this accesses a nested node_modules
-verify.not.codeFixAvailable();
+verify.codeFix({
+    index: 0,
```
[the remark this answers is not in the document]
Comment (a-tarasyuk) [reply tests/cases/fourslash/importNameCodeFixNewImportNodeModules5.ts:22]: Done

File: src/services/codefixes/fixAddMissingMember.ts [hunk 3]
```diff
@@ -93,23 +110,56 @@ namespace ts.codefix {
...
+
+        if (isIdentifier(token) && isCallExpression(parent.parent)) {
+            const moduleDeclaration = find(symbol.declarations, isModuleDeclaration);
+            const moduleDeclarationSourceFile = moduleDeclaration?.getSourceFile();
+            if (moduleDeclaration && moduleDeclarationSourceFile && !program.isSourceFileFromExternalLibrary(moduleDeclarationSourceFile)) {
+                return { kind: InfoKind.Function, token, call: parent.parent, sourceFile, modifierFlags: ModifierFlags.Export, parentDeclaration: moduleDeclaration };
+            }
+
+            const moduleSourceFile = find(symbol.declarations, isSourceFile);
+            if (sourceFile.commonJsModuleIndicator) {
```
Comment (jessetrinity) [inline src/services/codefixes/fixAddMissingMember.ts:155]: Why doesn't it work for common JS modules?

File: src/services/codefixes/helpers.ts [hunk 4]
```diff
@@ -247,41 +247,70 @@ namespace ts.codefix {
...
     }
 
-    export function createMethodFromCallExpression(
+    export function createFunctionFromCallExpression<T extends FunctionDeclaration | MethodDeclaration>(
+        kind: T["kind"] | SyntaxKind.MethodDeclaration,
         context: CodeFixContextBase,
         importAdder: ImportAdder,
         call: CallExpression,
-        methodName: string,
+        name: Identifier | string,
```
Comment (jessetrinity) [inline src/services/codefixes/helpers.ts:255]: nit: make both `string` or `Identifier` for `function` and `method` versions.

File: src/services/codefixes/helpers.ts [hunk 5]
```diff
@@ -247,41 +247,70 @@ namespace ts.codefix {
         );
     }
 
-    export function createMethodFromCallExpression(
+    export function createFunctionFromCallExpression<T extends FunctionDeclaration | MethodDeclaration>(
```
Comment (jessetrinity) [inline src/services/codefixes/helpers.ts:250]: The type parameter and the `kind` look like they're only ever both `MethodDeclaration` or `FunctionDeclaration`, so this seems redundant. Seems like you could skip on the kind if you want to go the type parameter route or just do the type assertion / check afterward.

**Review by jessetrinity — COMMENTED** [review]
I left a couple of questions but otherwise looks great 👍

File: src/services/codefixes/fixAddMissingMember.ts (hunk 3 above)
Comment (a-tarasyuk) [reply src/services/codefixes/fixAddMissingMember.ts:155]: @jessetrinity The issue description contains only samples with modules, https://github.com/microsoft/TypeScript/issues/26217, I decided that this is a limitation for common js. I think just need to check if `importAdder` supports common JS modules.

**Review by jessetrinity — APPROVED** [review]

**Pull request merged**